uplink/src/collector/systemstats.rs

```diff
@@ -0,0 +1,546 @@
+            max_buf_size,
+            tx.clone(),
+        );
+        for proc in sys.processors().iter() {
+            let proc_name = proc.name().to_owned();
+            map.insert(proc_name.clone(), Processor::init(proc_name));
+        }
+        let processors = Processors { sequence: 0, map, stream };
+
+        let stream = Stream::dynamic_with_size(
+            "process_stats",
+            &config.project_id,
+            &config.device_id,
+            max_buf_size,
+            tx.clone(),
+        );
+        let processes = Processes { sequence: 0, map: HashMap::new(), stream };
+
+        let stream = Stream::dynamic_with_size(
+            "system_stats",
+            &config.project_id,
+            &config.device_id,
+            max_buf_size,
+            tx.clone(),
+        );
+        let system = SystemStats { stat: SystemStat::init(&sys), stream };
+
+        let timestamp = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_millis() as u64;
+
+        StatCollector { sys, system, config, processes, disks, networks, processors, timestamp }
+    }
+
+    /// Stat collector execution loop, sleeps for the duation of `config.stats.update_period` in seconds.
+    pub fn start(mut self) {
+        loop {
+            std::thread::sleep(Duration::from_secs(self.config.stats.update_period));
+            self.timestamp += self.config.stats.update_period;
+
+            if let Err(e) = self.update() {
+                error!("Faced error while refreshing system statistics: {}", e);
+                return;
+            };
+        }
+    }
+
+    /// Update system information values and increment sequence numbers, while sending to specific data streams.
+    fn update(&mut self) -> Result<(), Error> {
+        if let Err(e) = self.system.push(&self.sys, self.timestamp) {
+            error!("Couldn't send system stats: {}", e);
+        }
+        self.sys.refresh_memory();
+
+        // Refresh disk info
+        for disk_data in self.sys.disks() {
+            if let Err(e) = self.disks.push(disk_data, self.timestamp) {
+                error!("Couldn't send disk stats: {}", e);
+            }
+        }
+        self.sys.refresh_disks();
+        self.sys.refresh_disks_list();
+
+        // Refresh network byte rate info
+        for (net_name, net_data) in self.sys.networks() {
+            if let Err(e) = self.networks.push(net_name.to_owned(), net_data, self.timestamp) {
+                error!("Couldn't send network stats: {}", e);
+            }
+        }
+        self.sys.refresh_networks();
+        self.sys.refresh_networks_list();
+
+        // Refresh processor info
+        for proc_data in self.sys.processors().iter() {
+            if let Err(e) = self.processors.push(proc_data, self.timestamp) {
+                error!("Couldn't send processor stats: {}", e);
+            }
+        }
+        self.sys.refresh_cpu();
+
+        // Refresh processes info
+        for (&id, p) in self.sys.processes() {
```

Comment:
Is it possible to just get our interested process instead of iterating through all the processes?

Reply:
Looks like there is no direct method but we update process data with pid. We need to do this iteration only during init and when pid doesn't match process name. Otherwise just refresh our processes of interest

Reply:
Ok, I was considering the situation where a process might die and another instance of it restarts with a new pid, same name, different pids also.